src/elements/a-progressive-asset.js

```diff
@@ -0,0 +1,57 @@
+AFRAME.registerElement("a-progressive-asset", {
+  prototype: Object.create(AFRAME.ANode.prototype, {
+    createdCallback: {
+      value() {
+        this.data = null;
+        this.isAssetItem = true;
+        this.fileLoader = document.querySelector("a-assets").fileLoader;
+      }
+    },
+
+    attachedCallback: {
+      value() {
+        const self = this;
+        const fallbackSrc = this.getAttribute("src");
+        const highSrc = this.getAttribute("high-src");
+        const lowSrc = this.getAttribute("low-src");
+
+        let src = fallbackSrc;
+
+        if (AFRAME.quality === "high") {
+          src = highSrc;
+        } else if (AFRAME.quality === "low") {
+          src = lowSrc;
+        }
+
+        this.fileLoader.setResponseType(this.getAttribute("response-type") || inferResponseType(src));
+        this.fileLoader.load(
+          src,
+          function handleOnLoad(response) {
+            self.data = response;
```

Comment:
I realize this isn't a new issue, but since we now are controlling both the loader and the asset, we should probably skip calling `GLTFLoader.prototype.load` and instead just pass this data directly to `GLTFLoader.prototype.parse`. In practice it doesn't actually matter since its cached, but seems clearer what is actually going on.

Reply:
Hmm, do we want a-progressive-asset to be specific to gltf assets? We could load lower resolution textures on lower-end platforms.